packages/template-retail-react-app/app/utils/url.js

```diff
@@ -178,16 +251,17 @@ export const getPathWithLocale = (shortCode, opts = {}) => {
  * @returns {string}
  */
 export const homeUrlBuilder = (homeHref, options = {}) => {
-    const {locale, site} = options
+    const {locale, site, fillUrlTemplate} = options
     const defaultSite = getDefaultSite()
     const isDefaultLocaleOfDefaultSite =
         locale.alias === defaultSite.l10n.defaultLocale ||
         locale.id === defaultSite.l10n.defaultLocale
     const isDefaultSite = site.id === defaultSite.id || site.alias === defaultSite.alias
-    const updatedUrl = buildPathWithUrlConfig(homeHref, {
-        locale: isDefaultLocaleOfDefaultSite && isDefaultSite ? '' : locale.alias || locale.id,
-        site: isDefaultLocaleOfDefaultSite && isDefaultSite ? '' : site.alias || site.id
-    })
+    const updatedUrl = fillUrlTemplate(
+        homeHref,
+        isDefaultLocaleOfDefaultSite && isDefaultSite ? '' : site.alias || site.id,
+        isDefaultLocaleOfDefaultSite && isDefaultSite ? '' : locale.alias || locale.id
```

Comment:
Isn't the point of all that hard work in the implementation of `fillUrlTemplate` that we don't have to have specific logic like this when using it.

Reply:
The main point of the PR is to optimize server-side rendering with a minimal set of changes. 
We achieved server-side optimization by simplifying the logic of the method used to build URLs across the app. 

But I agree, we can extend the URL template logic a bit so it can also handle the default site and locale Home Url. 
Done. The `homeUrlBuilder` method is no longer needed, the Home URL case is now handled in `createUrlTemplate`.